sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceivedMessageContext.java

```diff
@@ -35,6 +36,31 @@
         this.message = null;
     }
 
+    /**
+     * Creates an instance where an error occurred such as session-lock-lost.
+     *
+     * @param sessionId Session id that the error occurred in.
+     * @param error AMQP exception that occurred in session.
+     * @param errorSource the source of the error.
+     */
+    ServiceBusReceivedMessageContext(String sessionId, Throwable error, ServiceBusErrorSource errorSource) {
+        this(sessionId,error);
+        this.errorSource = Objects.requireNonNull(errorSource, "'errorSource' cannot be null.");
+    }
+
+    /**
+     * Creates an instance where an error occurred such as session-lock-lost.
+     *
+     * @param error AMQP exception that occurred in session.
+     * @param errorSource the source of the error.
+     */
+    ServiceBusReceivedMessageContext(Throwable error, ServiceBusErrorSource errorSource) {
```

Comment:
Why do need this?

Reply:
We can just use the maximal overload.

Reply:
moved this to ServiceBusException as we discussed.